### DISCUSSION TITLE
Error when using computeWithDatabase

### DISCUSSION THREAD

**Discussion by chrislaurie-io** [simolus3/drift#2329] [Q&A]
I am trying to use the computeWithDatabase api but I am getting an The argument type 'MyDatabase Function()' can't be assigned to the parameter type 'MyDatabase Function(DatabaseConnection)'. error when using the example from [here](https://drift.simonbinder.eu/docs/advanced-features/isolates/)

```
     await database.computeWithDatabase(
      computation: (databse){},
      connect: MyDatabase.new, <<< error is here
     );

```

I have tried various solutions but I cannot seem to solve this. Does anyone know where I am going wrong?

**Comment by simolus3** [comment]
The constructor of `MyDatabase` probably looks like this in your case:

```dart
LazyDatabase _constructMyDatabase() {
 // ...
}

@DriftDatabase(...)
class MyDatabase {
  MyDatabase(): super(_constructMyDatabase())
}
```

To enable `computeWithDatabase`, drift needs to be able to construct another instance of your database with a different underlying connection, so you need a database constructor allowing this.

You could add such constructor to your database class:

```dart
@DriftDatabase(...)
class MyDatabase {
  MyDatabase.custom(DatabaseConnection connection): super(connection);
  MyDatabase(): super(_constructMyDatabase())
}
```

Then, you can use `connect: MyDatabase.custom` in  `computeWithDatabase`.
